### PR TITLE
Brand model selection

### PR DESCRIPTION
**Pull request by DarkerTH**
* Created AutoAPI service for easy API access;
* Implemented live-fetching of models based on selected brand;

### CODE REVIEW

File: src/AppBundle/Controller/Api/CarController.php [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+use AppBundle\Service\AutoAPI;
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\Method;
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\Route;
+use Symfony\Bundle\FrameworkBundle\Controller\Controller;
+use Symfony\Component\HttpFoundation\Response;
+
+class CarController extends Controller
+{
+
+    function __construct()
```
Comment (zyrukas) [inline src/AppBundle/Controller/Api/CarController.php]: maybe unnecessary?

File: src/AppBundle/Resources/scripts/main.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
+$(document).ready(function () {
+
+    $('.dropdown-item').on('click', function (e) {
+        console.log('wot');
```
Comment (zyrukas) [inline src/AppBundle/Resources/scripts/main.js]: (°ロ°)☝

File: src/AppBundle/Service/AutoAPI.php [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+     * @param string $method
+     * @param string $endpoint
+     * @param array $query
+     * @return mixed|\Psr\Http\Message\ResponseInterface
+     */
+    private function request($method, $endpoint, Array $query=[]){
+        $api_host = $this->getApiHost();
+        $api_key = $this->getApiKey();
+
+        $client = new Client([
```
Comment (DarkerTH) [inline src/AppBundle/Service/AutoAPI.php]: @zyrukas

File: src/AppBundle/Service/AutoAPI.php [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+        return $models->getBody();
+    }
+
+    /**
+     * @return mixed|\Psr\Http\Message\ResponseInterface
+     */
+    public function getBrands()
+    {
+        $brands = $this->request('GET', 'brands');
+        return $brands->getBody();
```
Comment (zyrukas) [inline src/AppBundle/Service/AutoAPI.php]: pries returna visad turi buti \n, berods pagal symfony standartus

**Comment by zyrukas** [comment]
+1

**Comment by niumis** [comment]
+1

**Pull request merged**